backend/Application/Interop/InteropBinding.cs

```diff
@@ -0,0 +1,131 @@
+using System.Runtime.InteropServices;
+// ReSharper disable InconsistentNaming
+// Disabling are because names should follow the names in rust code. 
+
+namespace Application.Interop;
+
+/// <summary>
+/// Contains FFI bindings.
+/// </summary>
+internal static class InteropBinding
+{
+    private const string DllName = "librust_bindings";
+    
+    [DllImport(DllName, CallingConvention = CallingConvention.Cdecl, EntryPoint = "schema_display")]
+    private static extern bool schema_display(string schema, FFIOption schema_version, ref IntPtr result);
+    
+    [DllImport(DllName, CallingConvention = CallingConvention.Cdecl, EntryPoint = "get_receive_contract_parameter")]
+    private static extern bool get_receive_contract_parameter(string schema, FFIOption schema_version, string contract_name, string entrypoint, string value, ref IntPtr result);
+    
+    [DllImport(DllName, CallingConvention = CallingConvention.Cdecl, EntryPoint = "get_event_contract")]
+    private static extern bool get_event_contract(string schema, FFIOption schema_version, string contract_name, string value, ref IntPtr result);
+    
+    /// <summary>
+    /// Get module schema in a human interpretable form.
+    /// </summary>
+    /// <param name="schema">Module schema in hexadecimal</param>
+    /// <param name="schemaVersion">Optional schema version if present from module</param> 
+    /// <returns>Module schema in a human interpretable form</returns>
+    internal static InteropResult SchemaDisplay(string schema, FFIOption schemaVersion)
+    {
+        var result = IntPtr.Zero;
+        try
+        {
+            var schemaDisplay = schema_display(schema, schemaVersion, ref result);
+            var ptrToStringAnsi = Marshal.PtrToStringAnsi(result);
+            return new InteropResult(ptrToStringAnsi, schemaDisplay);
+        }
+        finally
+        {
+            Free(result);            
+        }
+    }
+    
+    /// <summary>
+    /// Get contract receive parameters in a human interpretable form.
+    ///
+    /// Receive parameters are those given to a contract entrypoint on a update call.
+    /// </summary>
+    /// <param name="schema">Module schema in hexadecimal</param>
+    /// <param name="contractName">Contract name</param>
+    /// <param name="entrypoint">Entrypoint of contract</param>
+    /// <param name="value">Receive parameters in hexadecimal</param>
+    /// <param name="schemaVersion">Optional schema version if present from module</param> 
+    /// <returns>Receive parameters in a human interpretable form</returns>
+    internal static InteropResult GetReceiveContractParameter(string schema, string contractName, string entrypoint, string value, FFIOption schemaVersion)
```

Comment:
I would take `schemaVersion` as a non-FFI type and do the wrapping in here to avoid leaking the low level types. Same in other cases.